src/fft.jl

```diff
@@ -108,13 +112,12 @@ function ifft!(f_real::AbstractArray3, fft_grid::FFTGrid, f_fourier::AbstractArr
     f_real .*= fft_grid.ifft_normalization
 end
 function ifft!(f_real::AbstractArray3, fft_grid::FFTGrid,
-               Gvec_mapping::Vector{Int}, f_fourier::AbstractVector; normalize=true)
-    @assert length(f_fourier) == length(Gvec_mapping)
-    @assert size(f_real) == fft_grid.fft_size
+               Gvec_mapping::AbstractVector{Int}, f_fourier::AbstractVector;
+               normalize=true)
 
     # Pad the input data
     fill!(f_real, 0)
-    f_real[Gvec_mapping] = f_fourier
+    ifft_remap!(f_real, Gvec_mapping, f_fourier)
```

Comment:
More meaningful names would be good. Something like fill_grid_with_kpt_data!()? Maybe also comment that this does `f_real[Gvec_mapping]=f_fourier`?